File: code_vita/QuestionD.py
def max_e( n, conflicts, expertise):
    max_expertise = 0 
    for mask in range(1 << n):
        valid_team = True
        expertise_sum= 0
        for i in range(n):
            if mask & (1 << i):
                expertise_sum += expertise[i]
                for j in range(n):
                    if mask & (1<<j) and conflicts[i][j]:
                        valid_team = False
                        break
                if not valid_team:
                    break
        if valid_team:
            max_expertise = max(max_expertise, expertise_sum)
    return max_expertise
n , c =  map(int, input().split())
conflicts = [[False] * n for _ in range(n)]

for _ in range(c):
    a, b = map(int, input().split())
    conflicts[a - 1][b - 1] = True
    conflicts[b - 1][a - 1] = True
expertise_levels = list(map(int , input().split()))
res = max_e(n , conflicts, expertise_levels)
print(res)

File: code_vita/QuestionD.cpp
#include <iostream>
#include <vector>

using namespace std;

int maxExpertise(int n, vector<vector<bool>> conflicts, vector<int> expertise) {
    int maxExpertise = 0;

    for (int mask = 0; mask < (1 << n); mask++) {
        bool validTeam = true;
        int expertiseSum = 0;

        for (int i = 0; i < n; i++) {
            if (mask & (1 << i)) {
                expertiseSum += expertise[i];

                for (int j = i + 1; j < n; j++) {
                    if ((mask & (1 << j)) && conflicts[i][j]) {
                        validTeam = false;
                        break;
                    }
                }

                if (!validTeam) {
                    break;
                }
            }
        }

        if (validTeam) {
            maxExpertise = max(maxExpertise, expertiseSum);
        }
    }

    return maxExpertise;
}

int main() {
    int n, c;
    cin >> n >> c;

    vector<vector<bool>> conflicts(n, vector<bool>(n, false));

    for (int i = 0; i < c; i++) {
        int a, b;
        cin >> a >> b;

        conflicts[a - 1][b - 1] = true;
        conflicts[b - 1][a - 1] = true;
    }

    vector<int> expertiseLevels(n);
    for (int i = 0; i < n; i++) {
        cin >> expertiseLevels[i];
    }

    int res = maxExpertise(n, conflicts, expertiseLevels);
    cout << res << endl;

    return 0;
}